collector/workitem.go

```diff
@@ -0,0 +1,337 @@
+	}
+	values["workitem"] = wi
+	users = append(users, collectWorkItemUsers(wi)...)
+
+	ownerID, _ := uuid.FromString(*wi.Data.Relationships.Creator.Data.ID)
+	workitemOwner, err := wit.GetUser(ctx, c, ownerID)
+	if err != nil {
+		errors = append(errors, err)
+	}
+	values["workitemOwner"] = workitemOwner
+
+	cs, err := wit.GetComments(ctx, c, wiID)
+	if err != nil {
+		errors = append(errors, err)
+	}
+	users = append(users, collectCommentsUsers(cs)...)
+
+	spaceID := *wi.Data.Relationships.Space.Data.ID
+	s, err := wit.GetSpace(ctx, c, spaceID)
+	if err != nil {
+		errors = append(errors, err)
+	}
+	values["space"] = s
+	users = append(users, collectSpaceUsers(s)...)
+
+	spaceOwner, err := wit.GetUser(ctx, c, *s.Data.Relationships.OwnedBy.Data.ID)
+	if err != nil {
+		errors = append(errors, err)
+	}
+	values["spaceOwner"] = spaceOwner
+
+	witype, err := wit.GetWorkItemType(ctx, c, spaceID, wi.Data.Relationships.BaseType.Data.ID)
+	if err != nil {
+		errors = append(errors, err)
+	}
+	values["workitemType"] = witype
+
+	actorID, err := getActorID(ctx)
+	if err == nil {
+		actor, err := wit.GetUser(ctx, c, actorID)
+		if err != nil {
+			errors = append(errors, err)
+		}
+		values["actor"] = actor
+	}
+
+	sc, err := wit.GetSpaceCollaborators(ctx, c, spaceID)
+	if err != nil {
+		errors = append(errors, err)
+	}
+	users = append(users, collectSpaceCollaboratorUsers(sc)...)
+
+	resolved, err := resolveAllUsers(ctx, c, SliceUniq(users), sc.Data)
+	if err != nil {
+		errors = append(errors, err)
+	}
+
+	if len(errors) > 0 {
+		return resolved, values, multiError{Message: "Errors during notification resolving", Errors: errors}
+	}
+
+	return resolved, values, nil
+}
+
+func resolveAllUsers(ctx context.Context, c *api.Client, users []uuid.UUID, collaborators []*api.UserData) ([]Receiver, error) {
+	var resolved []Receiver
+
+	for _, u := range users {
+		found := false
+		for _, c := range collaborators {
+			if u.String() == *c.ID {
+				found = true
+				if c.Attributes.Email != nil {
+					user := Receiver{EMail: *c.Attributes.Email}
+					if c.Attributes.FullName != nil {
+						user.FullName = *c.Attributes.FullName
+					}
+					resolved = append(resolved, user)
+				}
+			}
+		}
+		if !found {
+			c, err := wit.GetUser(ctx, c, u)
+			if err == nil {
+				if c.Data.Attributes.Email != nil {
+					user := Receiver{EMail: *c.Data.Attributes.Email}
+					if c.Data.Attributes.FullName != nil {
+						user.FullName = *c.Data.Attributes.FullName
+					}
+					resolved = append(resolved, user)
+				}
+			}
+		}
+	}
+
+	return resolved, nil
+}
+
+func collectSpaceCollaboratorUsers(cl *api.UserList) []uuid.UUID {
+	var users []uuid.UUID
+	for _, c := range cl.Data {
+		cID, err := uuid.FromString(*c.ID)
+		if err == nil {
+			users = append(users, cID)
+		}
+	}
+	return users
+}
+
+func collectCommentsUsers(cl *api.CommentList) []uuid.UUID {
+	var users []uuid.UUID
+
+	for _, c := range cl.Data {
+		if c.Relationships.CreatedBy != nil {
```

Comment:
Yeah, I saw that. And with no migration path in the API. @DhritiShikhar 

API's can't just change out of the blue and expect _everyone_ to change on the spot in the same second the WIT deploys. Wit should support both names for some time